src/ProtocolStreamWriter.php

```diff
@@ -25,6 +25,16 @@ public function __construct($output)
      */
     public function write(Message $msg)
     {
-        fwrite($this->output, (string)$msg);
+        $data = (string)$msg;
+        $msgSize = strlen($data);
+        $totalBytesWritten = 0;
+
+        while ($totalBytesWritten < $msgSize) {
+            $bytesWritten = @fwrite($this->output, substr($data, $totalBytesWritten));
+            if ($bytesWritten === false) {
+                throw new Error('Could not write message.');
+            }
+            $totalBytesWritten += $bytesWritten;
+        }
```

Comment:
If we write more data than the output buffer can hold we get partial writes, so we loop until all data is written.
This didn't happen before because in blocking mode this loop is handled internally.

We need to do this synchronously because otherwise we could end up with partial messages intermingled with each other.

Will change to `Exception`